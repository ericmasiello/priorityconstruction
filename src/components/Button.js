import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import tinyColor from 'tinycolor2';
import { COLORS, BODY_WEIGHTS } from '../styles/vars';

export const Button = ({ tag: Tag, ...rest }) => <Tag {...rest} />;

Button.propTypes = {
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
};

Button.defaultProps = {
  tag: 'button',
};

Button.displayName = 'Button';

export default styled(Button)`
  border: 1px solid ${COLORS.base};
  font-weight: ${BODY_WEIGHTS.bold};
  background-color: ${COLORS.bg};
  text-transform: uppercase;
  padding: 0.5rem 1.5rem;
  transition: background-color 0.2s;

  &:hover, &:focus {
    background-color: ${tinyColor(COLORS.bg)
    .darken(4)
    .toRgbString()};
  }
`;
