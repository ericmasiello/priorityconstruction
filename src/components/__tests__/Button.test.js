import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Button, { Button as Raw } from '../Button';

it('should render', () => {
  const component = renderer.create(<Button />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<Raw>Test</Raw>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<Raw data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<Raw className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <button> by default', () => {
  const wrapper = shallow(<Raw />);

  expect(wrapper.type()).toBe('button');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<Raw tag="span" />);

  expect(wrapper.type()).toBe('span');
});
