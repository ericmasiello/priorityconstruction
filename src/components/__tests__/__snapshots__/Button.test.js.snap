// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  border: 1px solid rgb(117,116,116);
  font-weight: 500;
  background-color: #fff;
  border-radius: 2px;
  text-transform: uppercase;
  padding: 0.5rem 1.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  min-height: 2.375rem;
}

.c0:hover,
.c0:focus {
  border-color: #424141;
  background-color: rgba(230,230,230,0.5);
}

.c0:disabled {
  pointer-events: none;
  border-color: rgb(196,194,194);
  color: #787575;
  background-color: rgb(221,220,220);
}

<button
  className="c0"
/>
`;
